cordapp-example/workflows-kotlin/src/main/kotlin/com/example/utils/FilterByState.kt

```diff
@@ -0,0 +1,13 @@
+package com.example.utils
+
+import net.corda.core.contracts.ContractState
+
+inline fun <reified T> List<ContractState>.filterByState(): List<T> {
+    val states = mutableListOf<T>()
+    for (element in this) {
+        if (element is T) {
+            states.add(element)
+        }
+    }
+    return states
+}
```

Comment:
A proper vault query already allows you to tell which class you want.

Reply:
I'm using this, to filter `ContractStates` that I already have from `stx.tx.inputs`. So is there a sense to query vault for something that I have?